### PR TITLE
Allows QIO and QOUT flash modes

### PR DESCRIPTION
**Pull request by wmrsouza**
Allows QIO and QOUT flash modes to work in:
- esp32s2
- esp32s3
- esp32c2
- esp32c3
- esp32c6

### CODE REVIEW

**Comment by EricNRS** [comment]
@wmrsouza  - thanks for taking a look at this issue.  I had tried enabling the flash wrap back in June and ran into a problem with the flash chip ID functionality.  Has that been fixed, now?  See https://github.com/zephyrproject-rtos/zephyr/issues/73677#issuecomment-2160147761 for details on the issue I ran into.

**Comment by wmrsouza** [comment]
Unfortunately, it's not fixed yet. I'm debugging the issue and will open this PR for review once it's working.

**Comment by wmrsouza** [comment]
@EricNRS

File: zephyr/esp32s3/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,8 @@ if(CONFIG_SOC_SERIES_ESP32S3)
   zephyr_compile_options(-fstrict-volatile-bitfields)
   zephyr_compile_definitions_ifndef(CONFIG_MCUBOOT CONFIG_APP_BUILD_USE_FLASH_SECTIONS)
   zephyr_compile_definitions_ifdef(CONFIG_MCUBOOT BOOTLOADER_BUILD)
+  zephyr_compile_definitions_ifdef(CONFIG_ESP_SIMPLE_BOOT AND CONFIG_ESPTOOLPY_FLASHMODE_QIO BOOTLOADER_BUILD)
```
Comment (sylvioalves) [inline zephyr/esp32s3/CMakeLists.txt:8]: I this case why not just leaving BOOTLOADER_BUILD enabled?
`zephyr_compile_definitions_ifdef(CONFIG_ESP_SIMPLE_BOOT BOOTLOADER_BUILD)`

**Comment by sylvioalves** [comment]
@wmrsouza, move `Fixes #73677` to Zephyr's PR instead.

File: zephyr/esp32s3/CMakeLists.txt [hunk 2]
```diff
@@ -145,7 +147,7 @@ if(CONFIG_SOC_SERIES_ESP32S3)
     ../../components/spi_flash/spi_flash_hpm_enable.c
     ../../components/spi_flash/spi_flash_os_func_noos.c
     ../../components/spi_flash/spi_flash_os_func_app.c
-    # todo: ../../components/spi_flash/spi_flash_wrap.c
+    ../../components/spi_flash/spi_flash_wrap.c
```
Comment (EricNRS) [inline zephyr/esp32s3/CMakeLists.txt:150]: Looks like this is the only change in this commit which I had tried back in June (see https://github.com/zephyrproject-rtos/zephyr/issues/73677#issuecomment-2160147761).  If this works now, then it seems that the root cause was fixed in some other changes that have already been merged.

Comment (wmrsouza) [reply zephyr/esp32s3/CMakeLists.txt:150]: In fact, this change will only fix the issue together with this other PR being merged  on the Zephyr side:
https://github.com/zephyrproject-rtos/zephyr/pull/79286

**Review by raffarost — APPROVED** [review]

**Review by LucasTambor — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Pull request merged**